### PR TITLE
Change to github actions

### PR DESCRIPTION
**Pull request by mbj**
### Acceptance

* [x] Verify minimal github actions setup. Intentionally I do the minimum to avoid bitrot in the future. No caching, bundle install is fast enough.
* [x] Accept drop of ruby versions below 2.1. I think this is a reasonable choice given 2.1 is EOL since 2017. I went the route of least friction here. Letting the lowest ruby version being defined by the github actions support.
* [x] Remove the required travis build success from branch settings (I'm not a repo owner else I could do this by myself, fell free to add me here).
* [x] Add the github action builds to required checks in branch settings.
* [x] Verify its okay to drop RBX. my rationale is that its irrelevant these days in production ruby.

### CODE REVIEW

File: memoizable.gemspec [hunk 1]
```diff
@@ -17,6 +17,4 @@ Gem::Specification.new do |gem|
   gem.extra_rdoc_files = Dir['**/*.md']
 
   gem.add_runtime_dependency('thread_safe', '~> 0.3', '>= 0.3.4')
-
-  gem.add_development_dependency('bundler', '~> 1.7', '>= 1.7.9')
```
Comment (mbj) [inline memoizable.gemspec:21]: @dkubb I'd normally extract this to a separate commit. But as the build has bitrotted (we last touched this repo in 2014!) I have to do this in one step.

Rationale for removing it: Explicit bundler dependency is not required these days. And even if we added one given the low commit frequency of this repo we'll end up in having to bump it always.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,10 +1,9 @@
 # encoding: utf-8
 
-if RUBY_VERSION >= '1.9'
```
Comment (mbj) [inline spec/spec_helper.rb:3]: No current CI, especially github actions has build in support for rubies smaller than 1.9.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,10 +1,9 @@
...
-  require 'simplecov'
-  require 'coveralls'
+require 'simplecov'
 
-  SimpleCov.formatters = [SimpleCov::Formatter::HTMLFormatter, Coveralls::SimpleCov::Formatter]
+if RUBY_ENGINE.eql?('ruby')
```
Comment (mbj) [inline spec/spec_helper.rb:5]: In jruby the `simplecov` does not work OTTB. For now I decided to accept this.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        - ruby-2.3
+        - ruby-2.4
+        - ruby-2.5
+        - ruby-2.6
+        - ruby-2.7
+        - ruby-jruby
+        os: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v2
+    - uses: ruby/setup-ruby@v1
```
Comment (mbj) [inline .github/workflows/ci.yml:32]: I've intentionally not set-up caching here. I want to do the CI setup so it has the minimal chance of rusting.

Given the low frequency of commits to this repository this is probably the right call.

**Comment by mbj** [comment]
@dkubb This is ready for review.

**Review by dkubb — APPROVED** [review]

**Comment by mbj** [comment]
@dkubb Thanks!
